test/crypto/elliptic.js

```diff
@@ -161,99 +177,81 @@ module.exports = () => describe('Elliptic Curve Cryptography @lightweight', func
-      0x8B, 0x37, 0x83, 0xE7, 0xBC, 0x1C, 0x4C, 0x86,
-      0xCB, 0x3C, 0xD5, 0x9B, 0x68, 0xF0, 0x65, 0xEB,
-      0x3A, 0xB6, 0xD6, 0xA6, 0xCF, 0x85, 0x3D, 0xA9
-    ]);
-    const p384_s = new Uint8Array([
-      0x32, 0x85, 0x78, 0xCC, 0xEA, 0xC5, 0x22, 0x83,
-      0x10, 0x73, 0x1C, 0xCF, 0x10, 0x8A, 0x52, 0x11,
-      0x8E, 0x49, 0x9E, 0xCF, 0x7E, 0x17, 0x18, 0xC3,
-      0x11, 0x11, 0xBC, 0x0F, 0x6D, 0x98, 0xE2, 0x16,
-      0x68, 0x58, 0x23, 0x1D, 0x11, 0xEF, 0x3D, 0x21,
-      0x30, 0x75, 0x24, 0x39, 0x48, 0x89, 0x03, 0xDC
-    ]);
-    it('Valid signature', function (done) {
-      expect(verify_signature('p384', 8, p384_r, p384_s, p384_message, key_data.p384.pub))
-        .to.eventually.be.true.notify(done);
+    it('P-384 - Valid signature', async function () {
+      const p384_r = new Uint8Array([
+        0x9D, 0x07, 0xCA, 0xA5, 0x9F, 0xBE, 0xB8, 0x76,
+        0xA9, 0xB9, 0x66, 0x0F, 0xA0, 0x64, 0x70, 0x5D,
+        0xE6, 0x37, 0x40, 0x43, 0xD0, 0x8E, 0x40, 0xA8,
+        0x8B, 0x37, 0x83, 0xE7, 0xBC, 0x1C, 0x4C, 0x86,
+        0xCB, 0x3C, 0xD5, 0x9B, 0x68, 0xF0, 0x65, 0xEB,
+        0x3A, 0xB6, 0xD6, 0xA6, 0xCF, 0x85, 0x3D, 0xA9
+      ]);
+      const p384_s = new Uint8Array([
+        0x32, 0x85, 0x78, 0xCC, 0xEA, 0xC5, 0x22, 0x83,
+        0x10, 0x73, 0x1C, 0xCF, 0x10, 0x8A, 0x52, 0x11,
+        0x8E, 0x49, 0x9E, 0xCF, 0x7E, 0x17, 0x18, 0xC3,
+        0x11, 0x11, 0xBC, 0x0F, 0x6D, 0x98, 0xE2, 0x16,
+        0x68, 0x58, 0x23, 0x1D, 0x11, 0xEF, 0x3D, 0x21,
+        0x30, 0x75, 0x24, 0x39, 0x48, 0x89, 0x03, 0xDC
+      ]);
+      const p384_message = new Uint8Array([
+        0x00, 0x11, 0x22, 0x33, 0x44, 0x55, 0x66, 0x77,
+        0x88, 0x99, 0xAA, 0xBB, 0xCC, 0xDD, 0xEE, 0xFF
+      ]);
+
+      await testNativeAndFallback(
+        () => expect(verify_signature('p384', 8, p384_r, p384_s, p384_message, key_data.p384.pub)).to.eventually.be.true
+      );
     });
-    it('Sign and verify message', function () {
-      const curve = new elliptic_curves.CurveWithOID('p521');
-      return curve.genKeyPair().then(async keyPair => {
-        const keyPublic = new Uint8Array(keyPair.publicKey);
-        const keyPrivate = new Uint8Array(keyPair.privateKey);
-        const oid = curve.oid;
-        const message = p384_message;
-        return elliptic_curves.ecdsa.sign(oid, 10, message, keyPublic, keyPrivate, await hashMod.digest(10, message)).then(async signature => {
-          await expect(elliptic_curves.ecdsa.verify(oid, 10, signature, message, keyPublic, await hashMod.digest(10, message)))
-            .to.eventually.be.true;
-        });
+    const curves = ['secp256k1' , 'p256', 'p384', 'p521', 'brainpoolP256r1', 'brainpoolP384r1', 'brainpoolP512r1'];
+    curves.forEach(curveName => it(`${curveName} - Sign and verify message`, async function () {
+      const curve = new elliptic_curves.CurveWithOID(curveName);
+      const { publicKey: keyPublic, privateKey: keyPrivate } = await curve.genKeyPair();
```

Comment:
```suggestion
      const { publicKey: keyPublic, privateKey: keyPrivate } = await curve.genKeyPair();
```
`genKeyPair()` does not pad the returned values. This causes random test failures in Node as some secret keys are 1 byte short. We should use `elliptic.generate` instead.